### PR TITLE
Move to circleci

### LINKED ISSUE #927: New microsite does not deploy as part of release
Microsite is built now using Docusaurus, the task appears to complete in Travis, but I did not investigate beyond that.

### LINKED ISSUE #931: Migrate build to Circle CI
Travis CI is slowly dying.
Also, other ZIO projects use Circle CI.

relates to #927  #767

### PR DESCRIPTION
**Pull request by softinio**
# Summary  

- Move from `travis` to `circleci`
- Microsite to publish to GH pages with Netlify pointing to GH pages

# Resolves 

resolves #931 
resolves #927 

# To Do 

### I will tackle post this PR:

#930 - **already resolved by** #943  

### New Issues to do later

#938
#944

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      - image: hseeberger/scala-sbt:8u181_2.12.8_1.2.8
+    steps:
+      - checkout
+      - run: git fetch --tags
+      - restore_cache:
+          key: sbt-cache
+      - run: echo -n "${PGP_PUBLIC}" | base64 -d > /tmp/public.asc
+      - run: echo -n "${PGP_SECRET}" | base64 -d > /tmp/secret.asc
+      - run: sbt releaseEarly
+  microsite:
```
Comment (LGLO) [inline .circleci/config.yml]: Minor/nitpic - we don't use microsite

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,8 @@ test-output/
 .sbtopts
 project/.sbt
 test-output/
+.bloop
```
Comment (LGLO) [inline .gitignore]: I think that good place for this is in  ~/.config/git/ignore . Likewise for `.idea/**`, `*.ipr`, etc.

File: .circleci/config.yml (hunk 1 above)
Comment (softinio) [reply .circleci/config.yml]: just the name of the step :-)

File: .gitignore (hunk 2 above)
Comment (softinio) [reply .gitignore]: I already have that in my ~/.config thought some value having it there for anyone who doesn't. metals and bloop are so popular now in my opinion it should be part of any default gitignore

**Comment by jdegoes** [comment]
@softinio Wow. Awesome work! 🎉

**Comment by softinio** [comment]
@jdegoes Thank you sir!

@NeQuissimus @ghostdogpr wait on review build currently affected by what seems to be this scalafmt related issue which I am investigating: 

https://github.com/scalameta/scalafmt/issues/1399

There is a workaround mentioned in the above issue if I don't find a better solution may have to go with it for now we will see. 

Will `@` you here once passed it and builds are successful.

**Comment by softinio** [comment]
@NeQuissimus @ghostdogpr RESOLVED! We have a successful build. Be wonderful if you can review today so we can complete the move to CircleCI today. 

cc @jdegoes 🎉

**Comment by jdegoes** [comment]
@softinio Superb work! If no one volunteers to review, let's merge and we can ask forgiveness later. 😉 

So much stuff is blocking on this that pushing it through and fixing up later is probably better than waiting.

**Comment by softinio** [comment]
@jdegoes Good idea will merge and tackle any further work necessary as separate PRs.

Created this to remove workaround for scalafmt issue once they fix. Following those PRs mentioned to be prompted to take action:

#944

**Pull request merged**